pinot-core/src/test/java/com/linkedin/pinot/queries/QueriesSentinelTest.java

```diff
@@ -149,7 +148,13 @@ private void runApproximationQueries(List<? extends AvroQueryGenerator.TestAggre
         if (query instanceof AvroQueryGenerator.TestSimpleAggreationQuery) {
           TestUtils.assertApproximation((Double) val, (Double) accurateValue, precision);
         } else {
-          TestUtils.assertJSONArrayApproximation((JSONArray) val, (JSONArray) accurateValue, precision);
+          List<GroupByResult> expectedValues = (List<GroupByResult>) accurateValue;
+          List<GroupByResult> groupByResults = (List<GroupByResult>) val;
+          for (int i = 0; i < groupByResults.size(); i++) {
+            double expected = Double.valueOf(groupByResults.get(i).getValue().toString());
+            Double actual = Double.valueOf(expectedValues.get(i).getValue().toString());
+            TestUtils.assertApproximation(actual, expected, precision);
```

Comment:
Seems like the too are not exactly same. Using TestUtils passes the test whereas using Assert does not. We can investigate the reason for that outside of this PR.